src/interaction.ts

```diff
@@ -18,92 +16,82 @@ import { signatureNoSighashType } from "unchained-bitcoin";
 /**
  * Constant representing a keystore which is unsupported due to the
  * kind of interaction or combination of paramters provided.
- *
- * @type {string}
  */
 export const UNSUPPORTED = "unsupported";
 
 /**
  * Constant representing a keystore pending activation by the user.
- *
- * @type {string}
  */
 export const PENDING = "pending";
 
 /**
  * Constant representing a keystore in active use.
- *
- * @type {string}
  */
 export const ACTIVE = "active";
 
 /**
  * Constant for messages at the "info" level.
- *
- * @type {string}
  */
 export const INFO = "info";
 
 /**
  * Constant for messages at the "warning" level.
- *
- * @type {string}
  */
 export const WARNING = "warning";
 
 /**
  * Constant for messages at the "error" level.
- *
- * @type {string}
  */
 export const ERROR = "error";
 
 /**
  * Enumeration of possible keystore states ([PENDING]{@link module:interaction.PENDING}|[ACTIVE]{@link module:interaction.ACTIVE}|[UNSUPPORTED]{@link module:interaction.UNSUPPORTED}).
  *
- * @constant
- * @enum {string}
- * @default
- *
  */
 export const STATES = [PENDING, ACTIVE, UNSUPPORTED];
 
 /**
  * Enumeration of possible message levels ([INFO]{@link module:interaction.INFO}|[WARNING]{@link module:interaction.WARNING}|[ERROR]{@link module:interaction.ERROR}).
- *
- * @constant
- * @enum {string}
- * @default
- *
  */
 export const LEVELS = [INFO, WARNING, ERROR];
 
 /**
- * Represents an image in a message returned by an interaction.
+ * Represents a message returned by an interaction.
  *
- * @typedef module:interaction.MessageImage
- * @type {Object}
- * @property {string} label - a human-readable label for the image
- * @property {string} mimeType - the MIME type of the image
- * @property {string} data - base64-encoded image data
+ * Message objects may have additional properties.
  */
+export type Message = {
+  state?: typeof STATES extends readonly (infer ElementType)[]
+    ? ElementType
+    : never;
```

Comment:
This was some black magic to me so I had to research a bit. In the off chance that anyone else is reading along: https://blog.logrocket.com/understanding-infer-typescript/

`typeof STATES extends` kind of creates a conditional checking if `STATES` is a readonly array. If it is, infer the type and assign it to `ElementType` and set that as the type, otherwise the type is `never`

(I think that's what's happening. I rewrote to help my own understanding)